src/lexer.coffee

```diff
@@ -1294,6 +1302,7 @@ POSSIBLY_DIVISION   = /// ^ /=?\s ///
 HERECOMMENT_ILLEGAL = /\*\//
 
 LINE_CONTINUER      = /// ^ \s* (?: , | \??\.(?![.\d]) | \??:: ) ///
+INDENT_SUPPRESSOR   = /// ^ \s* (?: and\s+(?!:)\S | or\s+(?!:)\S | && | \|\| ) ///
```

Comment:
The code could use a nice detailed comment explaining these two variables. Not the regexes themselves necessarily, though that wouldn’t hurt, but what a line continuer is and what `INDENT_SUPPRESSOR` is for (in terms of when and why you would suppress indents, of course).

Reply:
@GeoffreyBooth sure, added comments for `LINE_CONTINUER` and `INDENT_SUPPRESSOR` - the comments I added explain the token mechanics, but not sure if you think there should also be eg examples of what syntax these regexes allow?